src/tests/Microsoft.PowerFx.Core.Tests/TypeSystemTests/TypeTests.cs

```diff
@@ -82,5 +84,79 @@ public void TryGetFieldTypeLookupNullTest()
             Assert.Throws<ArgumentException>(
                 () => r1.TryGetFieldType(string.Empty, out var actualLogical, out var formulaType));
         }
+
+        [Theory]
+        [InlineData("X")]
+        [InlineData("$\"Test {X}\"")]
+        [InlineData("!X")]
+        [InlineData("Table(X)")]
+        [InlineData("X.Field1")]
+        [InlineData("X + 1")]
+        [InlineData("X + Date(2022, 10, 27)")]
+        [InlineData("X * 1")]
+        [InlineData("X < 2")]
+        [InlineData("\"The\" in X")]
+        [InlineData("123 in X")]
+        [InlineData("123 in Table(X)")]
+        [InlineData("IsBlank(X)")]
+        [InlineData("IsError(X)")]
+        [InlineData("Text(X)")]
+        [InlineData("Value(X)")]
+        [InlineData("Boolean(X)")]
+        [InlineData("Index([1,2,3].Value, X)")]
+
+        // Ensures expression binds without any errors - but issues a warning for the deferred(unknown) type.
+        public void DeferredTypeTest(string script)
+        {
+            Preview.FeatureFlags.StringInterpolation = true;
+            var symbolTable = new SymbolTable();
+            symbolTable.AddVariable("X", FormulaType.Unknown);
+            TestDeferredTypeBindingWarning(script, symbolTable);
+        }
+
+        [Theory]
+        [InlineData("$\"Test {X} {R}\"")]
+        [InlineData("Table(X, N)")]
+        [InlineData("X.field + N.field")]
+        [InlineData("Index([1,2,3], X).missing")]
+
+        // Ensures expression issues an error if it exists, despite the deferred type.
+        public void DeferredTypeTest_Negative(string script)
```

Comment:
>DeferredTypeTest_Negative [](http://example.com/codeflow?start=20&length=25)

we could make the assert stronger here by passing in a fragment of the expected error message and ensuring that shows up.